populus/utils/contracts.py

```diff
@@ -140,19 +140,46 @@ def is_contract_name(value):
     SWARM_HASH_SUFFIX
 )
 
+PUSH20_OPCODE = "73"
+ADDRESS_OPCODE = "30"
+EQ_OPCODE = "14"
+EMBEDDED_ADDRESS_REGEX = (
+    '^' +
+    PUSH20_OPCODE +
+    "[0-9a-z]{40}" +
+    ADDRESS_OPCODE +
+    EQ_OPCODE
+)
+
+ADDRESS_REPLACEMENT = (
+    PUSH20_OPCODE +
+    "<" +
+    "-" * 9 +
+    "address-place-holder" +
+    "-" * 9 +
+    ">" +
+    ADDRESS_OPCODE +
+    EQ_OPCODE
+)
+
 
 def compare_bytecode(left, right):
     unprefixed_left = remove_0x_prefix(left)
     unprefixed_right = remove_0x_prefix(right)
 
-    norm_left = re.sub(EMBEDDED_SWARM_HASH_REGEX, SWARM_HASH_REPLACEMENT, unprefixed_left)
-    norm_right = re.sub(EMBEDDED_SWARM_HASH_REGEX, SWARM_HASH_REPLACEMENT, unprefixed_right)
+    norm_left, norm_right = unprefixed_left, unprefixed_right
+    for regex, replacement in [
+        (EMBEDDED_SWARM_HASH_REGEX, SWARM_HASH_REPLACEMENT),
+        (EMBEDDED_ADDRESS_REGEX, ADDRESS_REPLACEMENT)
+    ]:
+        norm_left = re.sub(regex, replacement, norm_left)
+        norm_right = re.sub(regex, replacement, norm_right)
```

Comment:
Totally doesn't need to be done, but, if you wanted to do this functionally:

```
from cytoolz import curry, compose

sub = curry(re.sub)
norm_pipeline = compose(
    sub(EMBEDDED_SWARM_HASH_REGEX, SWARM_HASH_REPLACEMENT),
    sub(EMBEDDED_ADDRESS_REGEX, ADDRESS_REPLACEMENT),
)
norm_left = norm_pipeline(unprefixed_left)
norm_right = norm_pipeline(unprefixed_right)
```

Just in case you want to do this sort of thing in the future without the mutation.